### DISCUSSION TITLE
Restrict/secure pck file

### DISCUSSION THREAD

**Discussion by LoneDespair** [godotengine/godot-proposals#4715] [Engine Core]
**Header**
Letting users have the full power of Godot to mod a game, is a really fantastic feature
but with great power comes bad actors

**Goal**
Be able to restrict what a pck/mod can do,
like only access/overwrite files inside a specific directory, networking and/or etc

**Can this be implemented by a script?**
Technically yes
Practically no

**Comment by Zireael07** [comment]
I'm almost sure there's already a full blown proposal for that

**Comment by Calinou** [comment]
See https://github.com/godotengine/godot-proposals/issues/389 and https://github.com/godotengine/godot-proposals/issues/2689.

**Reply by anderlli0053** [reply]
Also obfuscating of PCK would be nice, so that cannot be easily extraced /reverse engineered (if so) by any kid...  even Unity has that...  and i am talking about whole scale here.. not just about the scripts, but all the assets inside the PCK (*.* "wildcard")

**Reply by Calinou** [reply]
This is possible in Godot 4.0: https://docs.godotengine.org/en/latest/development/compiling/compiling_with_script_encryption_key.html

That said, this kind of "protection" is always relative in practice. No matter the kind of encryption used, assets can be extracted as soon as you can hook into graphics API calls to save textures and models to disk. Sound can also be recorded from the game's audio output.
